### DISCUSSION TITLE
I have a project idea, but no resources or knowledge

### DISCUSSION THREAD

**Discussion by TuberculeP** [NMSCD/community#11] [General]
Hi there,
I have played NMS for a long time now and what I always found frustrating was how useful and broken reffining was, and how awful it was to look and learn all  possible recipes.

So, I began to code an algorithm capable of finding all refining ways for a given resource. I'm about to add some statistics for each recipes and resources to help finding the best way with a criteria (faster, more rentable, with common resources etc.)

Is there a database or a something like this to get the list of all resources ? with stats such as price, rarity, etc.
Maybe there is too for recipes but I can go with resources by default.

Also, I just discovered this place, I don't know if my idea is interesting for you guys and if it can be added to the main website or if I have to set it up on my own (I'm going to implement it in Vue3 front-only)

Thank you for all your work so far !

Félix

**Comment by Lenni009** [comment]
Hi! There is the [Assistant for NMS App](https://app.nmsassistant.com/) which has all resources and associated recipes. Other than that, you would have to dig through the game files yourself and write something to parse the EXML data into a format you can use.

**Comment by Khaoz-Topsy** [comment]
Sounds super cool @TuberculeP ! 
There is an npm package [assistantapps-nomanssky-info](https://www.npmjs.com/package/assistantapps-nomanssky-info) with almost all the game item data in it and localised for multiple languages, but it is quite large, probably not good for a front-end project 😅. Like @Lenni009 mentioned, The Assistant for No Man's Sky app has data on almost all the items in-game. You can look at this [folder](https://github.com/AssistantNMS/App/tree/main/assets/json/en) for the json files used in the apps 😋. You could use the Refinery.lang.json to get the inputs and output of each recipe with how long it takes to produce the output. The Ids that are used in this json file point to items listed in different json files. For example an item with the id `cur18` will be located in the Curiosity.lang.json file and an item with the id of `prod4` will be in the Products.lang.json file 😋. Feel free to ask for help understanding the data structure 😁

If you want the data in a specific way, then there is no better way than to extract it yourself from the game files as @Lenni009 mentioned 😋

We are open to anyone who wants to join the NMSCD, there aren't really any requirements and we don't claim any ownership of the repositories/projects. The organisation merely exists to try and increase collaboration 😋. Your project can definitely be added to the NMSCD organisation and we can host the website under a nmscd.com subdomain or folder. Up to you 😋. If your project is added to the NMSCD organisation, we will definitely add a link to it on the main website. Although with all the recent new repositories added (_cough cough_ @Lenni009 ), we will probably redesign the website so that it is easier to see all the projects we have. 😋

~ Kurt
